pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -388,85 +377,25 @@ func fixLibvirtCrcNetworkActive() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	logging.Debug("libvirt 'crc' network started")
-	return true, nil
-}
-
-func checkCrcDnsmasqConfigFile() (bool, error) {
-	logging.Debug("Checking dnsmasq configuration")
-	c := []byte(crcDnsmasqConfig)
-	_, err := os.Stat(crcDnsmasqConfigPath)
-	if err != nil {
-		return false, fmt.Errorf("File not found: %s: %s", crcDnsmasqConfigPath, err.Error())
-	}
-	config, err := ioutil.ReadFile(filepath.Clean(crcDnsmasqConfigPath))
-	if err != nil {
-		return false, fmt.Errorf("Error opening file: %s: %s", crcDnsmasqConfigPath, err.Error())
-	}
-	if !bytes.Equal(config, c) {
-		return false, fmt.Errorf("Config file contains changes: %s", crcDnsmasqConfigPath)
-	}
-	logging.Debug("dnsmasq configuration is good")
-	return true, nil
-}
-
-func fixCrcDnsmasqConfigFile() (bool, error) {
-	logging.Debug("Fixing dnsmasq configuration")
-	err := crcos.WriteToFileAsRoot(
-		fmt.Sprintf("write dnsmasq configuration in %s", crcDnsmasqConfigPath),
-		crcDnsmasqConfig,
-		crcDnsmasqConfigPath,
-	)
-	if err != nil {
-		return false, fmt.Errorf("Failed to write dnsmasq config file: %s: %v", crcDnsmasqConfigPath, err)
-	}
-
-	logging.Debug("Reloading NetworkManager")
-	sd := systemd.NewHostSystemdCommander()
-	if _, err := sd.Reload("NetworkManager"); err != nil {
-		return false, fmt.Errorf("Failed to restart NetworkManager: %v", err)
-	}
-
-	logging.Debug("dnsmasq configuration fixed")
-	return true, nil
-}
-
-func checkCrcNetworkManagerConfig() (bool, error) {
-	logging.Debug("Checking NetworkManager configuration")
-	c := []byte(crcNetworkManagerConfig)
-	_, err := os.Stat(crcNetworkManagerConfigPath)
-	if err != nil {
-		return false, fmt.Errorf("File not found: %s: %s", crcNetworkManagerConfigPath, err.Error())
-	}
-	config, err := ioutil.ReadFile(filepath.Clean(crcNetworkManagerConfigPath))
+	// Set DNS on the network
+	cmd = exec.Command("nmcli", "connection", "modify", "crc",
+		"ipv4.dns", "192.168.130.11",
+		"ipv4.dns-priority", "10")
```

Comment:
I had to run this command as root, otherwise it fails with "Insufficient privileges"

Reply:
Really? That's very strange cause here i don't need to elevate privileges. It kinda beats the point of this exercise if you need root access. :(

Reply:
@gbraad based on this and other hurdles, I don't really see a way forward here. I'd love to be wrong though. Shall I still try to think of ways this could be achieved or maybe I'm just missing some options that you'd have already thought of?

Reply:
we inch forward to a rootless setup. We can not expect all to be solved in day one. especially when even a fallback to hosts would still need root or raised permissions. this is a one time operation... however the driver using user session is a process that runs each single time. right?

Reply:
Setting the dnsmasq is a one time thing already so this won't help at all if it also requires root access, i-e we don't get any closer to non-root setup, which is why I thought perhaps this specific approach is not worth the trouble (I still haven't managed to make it work reliably or know why it won't work).

> however the driver using user session is a process that runs each single time. right?

Yes but i'm not sure about the relevance of that here. Did I miss something?